pkg/descheduler/strategies/duplicates.go

```diff
@@ -72,6 +74,39 @@ func deleteDuplicatePods(client clientset.Interface, policyGroupVersion string,
 	return podsEvicted
 }
 
+// computeCreatorSaturation finds if creators in the cluster are _saturated_.
+// A creator is _saturated_ if atleast one pod is running on every possible nodes.
+// In such a case, pods of this creator are not evicted from any nodes even if duplicates are present.
+func computeCreatorSaturation(client clientset.Interface, nodes []*v1.Node) (map[*v1.Node]DuplicatePodsMap, map[string]bool) {
+	dpmByNode := make(map[*v1.Node]DuplicatePodsMap)
+	creatorAssignedNodes := make(map[string][]*v1.Node)
+	for _, node := range nodes {
+		dpmByNode[node] = ListDuplicatePodsOnANode(client, node)
+		for creator := range dpmByNode[node] {
+			creatorAssignedNodes[creator] = append(creatorAssignedNodes[creator], node)
+		}
+	}
+
+	creatorPossibleNodes := make(map[string][]*v1.Node)
+	for creator, nodeList := range creatorAssignedNodes {
+		creatorNode := nodeList[0]
+		creatorPod := dpmByNode[creatorNode][creator][0]
+		for _, node := range nodes {
+			if nodeutil.PodFitsCurrentNode(creatorPod, node) && nodeutil.PodToleratesNodeTaints(creatorPod, node) {
```

Comment:
I think these 2 are not the only criteria for a pod to land on land, we should try to check all available predicates. IIUC, you want to evict pods only if the node has no duplicate on every node, so why can't we check if more than one copy exists on every node and if yes, don't do anything or else go with evicting the pod? To be clear, why just this check?